src/lib/datepicker/datepicker.ts

```diff
@@ -168,7 +168,18 @@ export class MatDatepicker<D> implements OnDestroy {
   @Output() selectedChanged = new EventEmitter<D>();
 
   /** Whether the calendar is open. */
-  opened = false;
+  @Input()
+  get opened(): boolean { return this._opened; }
+  set opened(value: boolean) {
+    const shouldOpen = coerceBooleanProperty(value);
```

Comment:
Since you're not assigning the `shouldOpen` anywhere, you don't really need to coerce it.